packages/liquidator/src/liquidator.js

```diff
@@ -118,18 +121,65 @@ class Liquidator {
           // Override must be one of the default logging levels: ['error','warn','info','http','verbose','debug','silly']
           return Object.values(overrides).every(param => Object.keys(this.logger.levels).includes(param));
         }
+      },
+      whaleDefenseFundWei: {
+        // by default make this disabled
+        value: undefined,
+        isValid: x => {
+          if (x === undefined) return true;
+          return this.toBN(x).gte(this.toBN("0"));
+        }
+      },
+      defenseActivationPercent: {
+        value: undefined,
+        isValid: x => {
+          if (x === undefined) return true;
+          return parseFloat(x) >= 0 && parseFloat(x) <= 100;
+        }
       }
     };
 
     // Validate and set config settings to class state.
-    Object.assign(this, createObjectFromDefaultProps(config, defaultConfig));
+    const configWithDefaults = createObjectFromDefaultProps(config, defaultConfig);
+    Object.assign(this, configWithDefaults);
+
+    // generalize log emitter, use it to attach default data to all logs
+    const log = (severity = "info", data = {}) => {
+      // would rather just throw here and let index.js capture and log, but
+      // its currently not set up to add in the additional context for the error
+      // so it has to be done here.
+      if (logger[severity] == null) {
+        return logger.error({
+          at: "Liquidator",
+          ...data,
+          error: "Trying to submit log with unknown severity: " + severity
+        });
+      }
+      return logger[severity]({
+        at: "Liquidator",
+        // could add in additional context for any error thrown,
+        // such as state of emp or bot configuration data
+        minLiquidationPrice: this.liquidationMinPrice,
+        ...data
+      });
+    };
+    this.log = log;
```

Comment:
this log function could be used in the rest of the liquidator file, but didnt want to refactor. It simply gets passed into liquidator strategy at this point. Because its a function its easy to mock and inspect in tests.